_sources/Functions/FunctionParameters.rst

```diff
@@ -0,0 +1,213 @@
+called **arguments** or **actual parameters** or **parameter values**, are passed to the function by the user.
+
+.. image:: Figures/blackboxproc.png
+
+This type of diagram is often called a **black-box diagram** because it only states the requirements from the 
+perspective of the user (well, the programmer, but the programmer who uses the function, who may be different than the 
+programmer who created the function). The user must know the name of the function and what arguments need to be 
+passed. The details of how the function works are hidden inside the "black-box".
+
+You have already been making function invocations with parameters. For example, when you write ``len("abc")`` or 
+``len([3, 9, "hello"])``, len is the name of a function, and the value that you put inside the parentheses, the string 
+"abc" or the list [3, 9, "hello"], is a parameter value.
+
+When a function has one or more parameters, the names of the parameters appear in the function definition, and the 
+values to assign to those parameters appear inside the parentheses of the function invocation. Let's look at each of 
+those a little more carefully.
+
+In the definition, the parameter list is sometimes referred to as the **formal parameters** or **parameter names**. 
+These names can be any valid variable name. If there is more than one, they are separated by commas. 
+
+In the function invocation, inside the parentheses one value should be provided for each of the parameter names. These 
+values are separated by commas. The values can be specified either directly, or by any python expression including a 
+reference to some other variable name.
+
+That can get kind of confusing, so let's start by looking at a function with just one parameter. The revised hello 
+function personalizes the greeting: the person to greet is specified by the parameter. 
+
+.. codelens:: clens11_3_1
+   :python: py3
+
+   def hello2(s):
+      print("Hello " + s)
+      print("Glad to meet you")
+         
+   hello2("Nick")
+   hello2("Jackie")
+
+First, notice that hello2 has one formal parameter, s. You can tell that because
+there is exactly one variable name inside the parentheses on line 1.
+
+Next, notice what happened during Step 2. Control was passed to the function, just like we saw before. But in 
+addition, the variable s was bound to a value, the string "Nick". When it got to Step 7, for the second invocation of 
+the function, s was bound to "Jackie".
+
+Function invocations always work that way. The expression inside the parentheses on the line that invokes the function 
+is evaluated before control is passed to the function. The value is assigned to the corresponding formal parameter. 
+Then, when the code block inside the function is executing, it can refer to that formal parameter and get its value, 
+the value that was 'passed into' the function.
+
+To get a feel for that, let's invoke hello2 using some more complicated expressions. Try some of your own, too.
```

Comment:
I think a showeval that demonstrates how the function parameters are evaluated before the function is called would be helpful here.